components/camel-google/camel-google-pubsub-lite/src/main/java/org/apache/camel/component/google/pubsublite/GooglePubsubLiteEndpoint.java

```diff
@@ -0,0 +1,238 @@
+public class GooglePubsubLiteEndpoint extends DefaultEndpoint {
+
+    private Logger log;
+
+    @UriPath(label = "common", description = "The Google Cloud PubSub Lite Project Id")
+    @Metadata(required = true)
+    private Long projectId;
+
+    @UriPath(label = "common", description = "The Google Cloud PubSub Lite location")
+    @Metadata(required = true)
+    private String location;
+
+    @UriPath(label = "common",
+             description = "The Destination Name. For the consumer this will be the subscription name, while for the producer this will be the topic name.")
+    @Metadata(required = true)
+    private String destinationName;
+
+    @UriParam(label = "common",
+              description = "The Service account key that can be used as credentials for the PubSub publisher/subscriber. It can be loaded by default from "
+                            + " classpath, but you can prefix with classpath:, file:, or http: to load the resource from different systems.")
+    private String serviceAccountKey;
+
+    @UriParam(name = "loggerId", description = "Logger ID to use when a match to the parent route required")
+    private String loggerId;
+
+    @UriParam(label = "consumer", name = "concurrentConsumers",
+              description = "The number of parallel streams consuming from the subscription",
+              defaultValue = "1")
+    private Integer concurrentConsumers = 1;
+
+    @UriParam(label = "consumer", name = "maxMessagesPerPoll",
+              description = "The max number of messages to receive from the server in a single API call", defaultValue = "1")
+    private Integer maxMessagesPerPoll = 1;
+
+    @UriParam(label = "consumer", defaultValue = "AUTO", enums = "AUTO,NONE",
+              description = "AUTO = exchange gets ack'ed/nack'ed on completion. NONE = downstream process has to ack/nack explicitly")
+    private GooglePubsubLiteConstants.AckMode ackMode = GooglePubsubLiteConstants.AckMode.AUTO;
+
+    @UriParam(label = "consumer", name = "maxAckExtensionPeriod",
+              description = "Set the maximum period a message ack deadline will be extended. Value in seconds",
+              defaultValue = "3600")
+    private int maxAckExtensionPeriod = 3600;
+
+    @UriParam(description = "Pub/Sub endpoint to use. Required when using message ordering, and ensures that messages are received in order even when multiple publishers are used",
+              label = "producer,advanced")
+    private String pubsubEndpoint;
+
+    @UriParam(name = "serializer",
+              description = "A custom GooglePubsubLiteSerializer to use for serializing message payloads in the producer",
+              label = "producer,advanced")
+    @Metadata(autowired = true)
+    private GooglePubsubSerializer serializer;
+
+    public GooglePubsubLiteEndpoint(String uri, Component component) {
+        super(uri, component);
+
+        if (!(component instanceof GooglePubsubLiteComponent)) {
```

Comment:
I think this is a bit overkilling, we could avoid this check.

Reply:
Fixed in [8bfe10f](https://github.com/apache/camel/pull/13630/commits/8bfe10f92a9fd68d9d9fa417c5e906aca3b20d7a)